packages/traveling-fastlane/actions/manage_ad_hoc_provisioning_profile.rb

```diff
@@ -0,0 +1,140 @@
+require 'spaceship'
+require 'json'
+require 'base64'
+require 'set'
+require_relative 'funcs'
+
+ENV['SPACESHIP_AVOID_XCODE_API'] = '1'
+
+$teamId, $udidsString, $bundleId, $certSerialNumber = ARGV
+$udids = $udidsString.split(',')
+
+$result = nil
+
+def find_dist_cert(serialNumber, isEnterprise)
+  certs = if isEnterprise == 'true'
+    Spaceship::Portal.certificate.in_house.all
+  else
+    Spaceship::Portal.certificate.production.all
+  end
+
+  if $certSerialNumber == '__last__'
+    certs.last
+  else
+    certs.find do |c|
+      c.raw_data['serialNum'] == $certSerialNumber
+    end
+  end
+end
+
+def register_missing_devices(udids)
+  all_iphones = Spaceship.device.all_iphones
+  already_added = all_iphones.select { |d| d.enabled? and udids.include?(d.udid) }
+  already_added_udids = already_added.map { |i| i.udid }
+
+  devices = [*already_added]
+
+  udids_to_add = udids - already_added_udids
+  udids_to_add.each { |udid|
+    devices.push Spaceship.device.create!(name: "iPhone (added by Expo)", udid: udid)
```

Comment:
i think the UDID service will get a device name to go along with the UDID, would be good to have that instead of this generic name. (but not necessary for this first iteration)